### PR TITLE
Fix EG samples

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Samples were failing based on new validation - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=751823&view=ms.vss-test-web.build-test-results-tab&runId=16879826&resultId=100018&paneView=debug

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/Samples/Sample1_SendEventsToTopicAndDomain.cs [hunk 1]
```diff
@@ -90,37 +90,43 @@ public async Task SendCloudEventsToTopic()
...
+                    myCustomDataSerializer.Serialize(new CustomModel() { A = 5, B = true }),
+                    "application/json"),
 
                 // CloudEvents also supports sending binary-valued data
                 new CloudEvent(
                     "/cloudevents/example/binarydata",
                     "Example.EventType",
-                    new BinaryData(Encoding.UTF8.GetBytes("This is binary data")),
+                    new BinaryData(Encoding.UTF8.GetBytes("This is treated as binary data")),
                     "example/binary")};
```
Comment (jsquire) [inline sdk/eventgrid/Azure.Messaging.EventGrid/tests/Samples/Sample1_SendEventsToTopicAndDomain.cs:130]: did you want to also update this to the octet-stream used elsewhere for the content-type of binary data?

**Review by jsquire — APPROVED** [review]

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/Samples/Sample1_SendEventsToTopicAndDomain.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/Samples/Sample1_SendEventsToTopicAndDomain.cs:130]: yes..

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/Samples/Sample1_SendEventsToTopicAndDomain.cs:130]: disabling auto-merge 😄

Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/Samples/Sample1_SendEventsToTopicAndDomain.cs]: I'm actually not sure how I managed that considering both md files were using the same snippet.

**Review by pakrym — APPROVED** [review]

**Pull request merged**